### PR TITLE
ENH: #8750 add Series support for to_html and _repr_html_

### PR DESCRIPTION
**Pull request by lexual**
closes #5563

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -926,6 +926,18 @@ def _repr_footer(self):
...
                                                str(self.dtype.name))
 
+    def _repr_html_(self, *args, **kwargs):
+        df = self.to_frame()
+        if self.name is None:
+            df.columns = ['']
+        return df._repr_html_(*args, **kwargs)
+
+    def to_html(self, *args, **kwargs):
+        df = self.to_frame()
```
Comment (sinhrks) [inline pandas/core/series.py]: Can this be a call of ``_repr_html_``?

File: pandas/core/series.py [hunk 2]
```diff
@@ -926,6 +926,18 @@ def _repr_footer(self):
                                                len(self),
                                                str(self.dtype.name))
 
+    def _repr_html_(self, *args, **kwargs):
+        df = self.to_frame()
+        if self.name is None:
+            df.columns = ['']
+        return df._repr_html_(*args, **kwargs)
+
+    def to_html(self, *args, **kwargs):
```
Comment (sinhrks) [inline pandas/core/series.py]: Please add the function to ``api.rst``, and better to use explicit keywords to render API doc.

File: pandas/core/series.py (hunk 1 above)
Comment (lexual) [inline pandas/core/series.py]: I have called DataFrame's _repr_html in Series _repr_html and DataFrame's to_html in Series to_html.

If you look at the test case they have 2 different behaviours/outcomes.

Do you recommend only calling _repr_html in both cases and the behaviour being inconsistent with the DataFrame behaviour?

Comment (sinhrks) [inline pandas/core/series.py]: Ah, sorry. I overlooked differences.

**Comment by shoyer** [comment]
@lexual Could you add a image showing what the new `_repr_html_` looks like for Series? The main challenge here is that we'd like to be sure it remains clearly visually distinct from that for DataFrame.

**Comment by shoyer** [comment]
Looking through your code, I see that you are basically just calling `s.to_frame()._repr_html()`. This is a bit of a non-starter -- we need something that is obviously different. Users should know when they're working with a series instead of a frame.

**Comment by jreback** [comment]
@lexual can you rebase & add a whatsnew note.

**Comment by jreback** [comment]
@lexual can you update according to comments

**Comment by jreback** [comment]
@lexual like to put this in 0.17.1. can you rebase/update

**Comment by jorisvandenbossche** [comment]
@jreback Given the last comment of @shoyer above (https://github.com/pydata/pandas/pull/10300#issuecomment-109693262), I don't think this is ready for inclusion in 0.17.1. We should first figure out how we want it to look like (needs to be a distinction from dataframe with one column)

File: pandas/core/series.py (hunk 2 above)
Comment (lexual) [inline pandas/core/series.py]: added

**Comment by lexual** [comment]
@jreback have rebased, let us know if merge-able, or needs reworking ;)

**Comment by jorisvandenbossche** [comment]
Can you show how the output looks like?

**Comment by jreback** [comment]
@lexual I think we *would* like to put this in, but I agree that it can't look like a 1-column frame, maybe same but no header?

**Comment by jreback** [comment]
@lexual can you show a render of this?

**Comment by jreback** [comment]
can you update

**Comment by jreback** [comment]
@lexual can you update?

**Comment by jreback** [comment]
@lexual can you rebase / update

**Comment by jreback** [comment]
can you rebase/update.

**Comment by jreback** [comment]
@lexual can you rebase / update

**Comment by jreback** [comment]
can you rebase / update?

**Comment by sinhrks** [comment]
It's nice if you can meet v0.19.0. Otherwise allow me to work this on.

**Comment by jorisvandenbossche** [comment]
Closing this, @sinhrks I think you can go ahead (or rebase this one). But see also the comments above that are not yet resolved I think from @shoyer and me:

> I see that you are basically just calling s.to_frame()._repr_html(). This is a bit of a non-starter -- we need something that is obviously different. Users should know when they're working with a series instead of a frame.

> We should first figure out how we want it to look like (needs to be a distinction from dataframe with one column)

**Comment by lexual** [comment]
Apologies haven't been able to respond or rebase as requested in the past.

Did this get merged, or just closed?

Does it need major surgery to get merged, or just rebase & possibly tidy up?

Thanks ;)

**Comment by shoyer** [comment]
I don't think @sinhrks has started on this yet, but he can tell you for sure. It has not been merged but was closed because it was stale. If you are interested in working on this again we can reopen.

The important thing to figure out first (maybe in the linked issue) is what the HTML repr should actually look like.

**Comment by ellisonbg** [comment]
The styling of HTML tables in the jupyter notebook is provided by the jupyter CSS itself. From this perspective, the HTMl repr simply needs to have a well formed table format with `tbody` and `thead` blocks. Should be very simple. Also we should test on the new table styling that is in master of jupyter and jupyterlab...

**Comment by jorisvandenbossche** [comment]
@ellisonbg Technically there is indeed no problem, just having `_repr_html` output a well behaved html snippet. 
The issue is more 'design'-related, as we do *not* want it to look just like a normal table, as in that case it would not distinguish well from a one-column dataframe. Maybe not that difficult, but up to now nobody put forward a concrete proposal for how it could look like.

**Pull request closed without merging**